src/test/scala/org/camunda/feel/impl/builtin/BuiltinConversionFunctionsTest.scala

```diff
@@ -207,6 +207,12 @@ class BuiltinConversionFunctionsTest
       ValDayTimeDuration("P2DT20H14M"))
   }
 
+  "A duration() function" should "parse string variables with -" in {
+
+    eval(""" duration(x) """, Map("x" -> "PT-5M")) should be(
+      ValDayTimeDuration("PT-5M"))
```

Comment:
:x: We should add more test cases to cover all possible options.

For example:

- `P-1DT-2H-3M-4S` - returns `null`
- `-P1DT2H3M4S` - works

Reply:
:wrench: We could add similar test cases for a years-months-duration string. See the test case below.

Reply:
:wrench: We could add similar test cases for the duration literals (i.e. `duration("-P1D")` vs `duration(x)`).

The test cases for the literals are in the class [InterpreterLiteralExpressionTest](https://github.com/camunda/feel-scala/blob/main/src/test/scala/org/camunda/feel/impl/interpreter/InterpreterLiteralExpressionTest.scala#L236-L262). Currently, they don't cover negative durations.

Reply:
Should the InterpreterTest be a follow up work item? Seems like that would be a different issue right?

Reply:
> Should the InterpreterTest be a follow up work item? Seems like that would be a different issue right?

Yes. This was not part of the reported issue. But the implementation uses the same code. 

Feel to adjust the `InterpreterLiteralExpressionTest` in a follow-up PR.